### PR TITLE
Enable-path-restrictions-for-Android-11

### PR DESCRIPTION
**Pull request by yasoda-aravapalli**
Tracked-On: OAM-95316
Signed-off-by: yaravapa <yasoda.aravapalli@intel.com>

### CODE REVIEW

**Review by shyjumon-n — APPROVED** [review]
Looks good to me

File: bsp_diff/common/kernel/lts2019-chromium/68_0068-Use-clang-toolchain-from-prebuilts-instead-of-gcc.patch [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
++#endif
+ KBUILD_HOSTCFLAGS   := -Wall -Wmissing-prototypes -Wstrict-prototypes -O2 \
+ 		-fomit-frame-pointer -std=gnu89 $(HOST_LFS_CFLAGS) \
+ 		$(HOSTCFLAGS)
+@@ -410,7 +410,7 @@ KBUILD_HOSTLDLIBS   := $(HOST_LFS_LIBS) $(HOSTLDLIBS)
+ 
+ # Make variables (CC, etc...)
+ CPP		= $(CC) -E
+-ifneq ($(LLVM),)
++#ifneq ($(LLVM),)
```
Comment (JeevakaPrabu) [inline bsp_diff/common/kernel/lts2019-chromium/68_0068-Use-clang-toolchain-from-prebuilts-instead-of-gcc.patch:42]: LLVM is set to 1. so, changes in Makefile is not needed.

File: aosp_diff/preliminary/build/soong/0007-Enable-path-restrictions-for-Android-11.patch [hunk 2]
```diff
@@ -0,0 +1,73 @@
+From 5ed3ae1e2e3ff649411dc05bfe6794446cc1e62a Mon Sep 17 00:00:00 2001
+From: yaravapa <yasoda.aravapalli@intel.com>
+Date: Mon, 25 Jan 2021 09:56:05 +0530
+Subject: [PATCH] Enable path restrictions for Android 11
+
+Tracked-On: OAM - 95316
```
Comment (JeevakaPrabu) [inline aosp_diff/preliminary/build/soong/0007-Enable-path-restrictions-for-Android-11.patch:6]: Change "OAM - 95316"  to "OAM-95316". No space needed

**Review by JeevakaPrabu — CHANGES_REQUESTED** [review]
Please remove "-" in between words "Enable-path-restrictions-for-Android-11"

File: bsp_diff/common/vendor/intel/fw/evmm/0001-Use-prebuilts-clang-instead-of-gcc-to-build-evmmpack.patch [hunk 3]
```diff
@@ -0,0 +1,55 @@
+From 09a06726a6d710c4411ffe5b7e9c448a0379379c Mon Sep 17 00:00:00 2001
```
Comment (YadongQi) [inline bsp_diff/common/vendor/intel/fw/evmm/0001-Use-prebuilts-clang-instead-of-gcc-to-build-evmmpack.patch:1]: why not submit to upstream? please PR to: https://github.com/intel/ikgt-core/tree/trusty

File: bsp_diff/common/vendor/intel/fw/evmm/0001-Use-prebuilts-clang-instead-of-gcc-to-build-evmmpack.patch [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ #define ALIGN_B(value, align) \
+ 	((uint64_t)(value) & (~((uint64_t)(align) - 1ULL)))
+diff --git a/packer/Makefile b/packer/Makefile
+index 3d5c88d..5392f55 100644
+--- a/packer/Makefile
++++ b/packer/Makefile
+@@ -7,7 +7,7 @@
+ ################################################
+ 
+ #using local gcc for evmmpacker
```
Comment (YadongQi) [inline bsp_diff/common/vendor/intel/fw/evmm/0001-Use-prebuilts-clang-instead-of-gcc-to-build-evmmpack.patch:39]: amend comments: using local clang for evmmpacker by default.

File: aosp_diff/preliminary/development/python-packages/0001-Adding-mako-module-to-python-development-packages.patch [hunk 5]
```diff
@@ -0,0 +1,8206 @@
+From eeb07e7b48b8c28f0c683a5b9d77fd57cae0ee0b Mon Sep 17 00:00:00 2001
+From: yaravapa <yasoda.aravapalli@intel.com>
+Date: Tue, 19 Jan 2021 16:35:46 +0530
+Subject: [PATCH] Adding mako module to python development packages
+
+This is required as part of enabling path restrictions
+in Android 11.
+Python scripts in hardware/intel/mesa3d are dependent on
```
Comment (renchenglei) [inline aosp_diff/preliminary/development/python-packages/0001-Adding-mako-module-to-python-development-packages.patch:8]: Mesa path should be hardware/intel/external/mesa3d-intel.

**Comment by renchenglei** [comment]
Thanks @yasoda-aravapalli. LGTM. Previously, we use host mako tool for mesa build?

**Comment by yasoda-aravapalli** [comment]
> Thanks @yasoda-aravapalli. LGTM. Previously, we use host mako tool for mesa build?

Yes previously we used mako tool from host

**Comment by renchenglei** [comment]
> > Thanks @yasoda-aravapalli. LGTM. Previously, we use host mako tool for mesa build?
> 
> Yes previously we used mako tool from host

Got it, thanks a lot for the work.

File: aosp_diff/preliminary/development/python-packages/0001-Adding-mako-module-to-python-development-packages.patch (hunk 5 above)
Comment (yasoda-aravapalli) [reply aosp_diff/preliminary/development/python-packages/0001-Adding-mako-module-to-python-development-packages.patch]: Updated the patch commit message with correct Mesa path hardware/intel/external/mesa3d-intel.

**Review by feijiang1 — APPROVED** [review]

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**